lib/services/metadata_provider.dart

```diff
@@ -49,182 +22,200 @@ class MetadataProvider {
           } catch (e) {
             metadataProviderLogger.warning(
-              "Failed to check if '${request.item.name}' (${request.item.id}) qualifies for playback speed controls",
+              "Failed to fetch lyrics for '${item.name}' (${item.id}). Metadata might be stale",
               e,
             );
           }
         }
       }
-    }
-  }
-
-  if (request.includeLyrics && metadata.hasLyrics) {
-    //!!! only use offline metadata if the app is in offline mode
-    // Finamp should always use the server metadata when online, if possible
-    if (FinampSettingsHelper.finampSettings.isOffline) {
-      DownloadedLyrics? downloadedLyrics;
-      downloadedLyrics = await downloadsService.getLyricsDownload(baseItem: request.item);
-      if (downloadedLyrics != null) {
-        metadata.lyrics = downloadedLyrics.lyricDto;
-        metadataProviderLogger.fine("Got offline lyrics for '${request.item.name}'");
-      } else {
-        metadataProviderLogger.fine("No offline lyrics for '${request.item.name}'");
-      }
-    } else {
-      metadataProviderLogger.fine("Fetching lyrics for '${request.item.name}' (${request.item.id})");
-      try {
-        final lyrics = await jellyfinApiHelper.getLyrics(itemId: request.item.id);
-        metadata.lyrics = lyrics;
-      } catch (e) {
-        metadataProviderLogger.warning(
-          "Failed to fetch lyrics for '${request.item.name}' (${request.item.id}). Metadata might be stale",
-          e,
-        );
-      }
-    }
-  }
 
-  metadataProviderLogger.fine(
-    "Fetched metadata for '${request.item.name}' (${request.item.id}): ${metadata.lyrics} ${metadata.hasLyrics}",
-  );
+      metadataProviderLogger.fine(
+        "Fetched metadata for '${item.name}' (${item.id}): ${metadata.lyrics} ${metadata.hasLyrics}",
+      );
+
+      return metadata;
+    });
+
+final AutoDisposeFutureProviderFamily<BaseItemDto?, BaseItemId> albumProvider = FutureProvider.autoDispose
+    .family<BaseItemDto?, BaseItemId>((ref, parentId) async {
+      final jellyfinApiHelper = GetIt.instance<JellyfinApiHelper>();
+      final downloadsService = GetIt.instance<DownloadsService>();
 
-  return metadata;
-});
+      if (ref.watch(finampSettingsProvider.isOffline)) {
+        final parentInfo = await downloadsService.getCollectionInfo(id: parentId);
+        if (parentInfo == null) {
+          metadataProviderLogger.warning("Couldn't find parent collection '$parentId' in offline mode");
+        } else if (parentInfo.baseItem == null) {
+          metadataProviderLogger.warning("Offline metadata for '$parentId' does not include jellyfin BaseItemDto");
+        } else {
+          return parentInfo.baseItem;
+        }
+      } else {
+        try {
+          return await jellyfinApiHelper.getItemById(parentId);
+        } catch (e) {
+          metadataProviderLogger.warning("Failed to get parent item '$parentId'", e);
+        }
+      }
+      return null;
+    });
```

Comment:
I think it might make sense to eventually group such providers together into one file/directory. Having them embedded in a "random" file is a bit strange, and I don't think this provider is actually specific to this screen (even if it's currently only used here).  

We could probably reduce the code/logic by quite a lot if we used providers for fetching all base items, and handling online/offline mode, even if the providers are non-caching.  
That's out-of-scope for this specific PR though :)